src/test/java/com/google/sps/servlets/DataServletTest.java

```diff
@@ -0,0 +1,162 @@
+package com.google.sps.servlets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.cloud.language.v1.AnalyzeSentimentResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class DataServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private DataServlet newTermRating;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    languageService = Mockito.mock(LanguageServiceClient.class);
+    newTermRating = new DataServlet(languageService);
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock LanguageServiceClient languageService;
+  @Mock HttpServletRequest request;
+
+  @Test
+  public void addTermRating_newRating() throws IOException {
+
+    // File with body request in webapp folder.
+    Reader reader = new FileReader("src/main/webapp/WEB-INF/testNewRating.txt");
+    BufferedReader bufferedReader = new BufferedReader(reader);
+    when(request.getReader()).thenReturn(bufferedReader);
+
+    AnalyzeSentimentResponse response =
+        AnalyzeSentimentResponse.newBuilder()
+            .setDocumentSentiment(Sentiment.newBuilder().setScore((float) -0.8999999761581421))
```

Comment:
Where does this number come from? If it's a constant referenced throughout you should extract it to a variable somewhere, if it's just random than put a note there that it's meaningless.